.editorconfig

```diff
@@ -1,32 +1,23 @@
 # top-most EditorConfig file
 root = true
 
+# ----------------------------------------------------------------------------------------------------------------------
+# General settings
+# ----------------------------------------------------------------------------------------------------------------------
+
 # Don't use tabs for indentation.
 [*]
-charset = utf-8
 indent_style = space
-insert_final_newline = false
-trim_trailing_whitespace = true
 # (Please don't specify an indent_size here; that has too many unintended consequences.)
 
-# Scripting files
-[*.{ps1,bat,cmd}]
-indent_size = 4
-
-# C# code files
-[*.{cs}]
+# Code files
+[*.{cs,csx,vb,vbx}]
 indent_size = 4
-
-# Java code files
-[*.{java}]
-indent_size = 2
-
-# Pom files
-[pom.xml]
-indent_size = 2
+insert_final_newline = true
+charset = utf-8-bom
 
 # Xml project files
-[*.{csproj,proj}]
+[*.{csproj,vbproj,vcxproj,vcxproj.filters,proj,projitems,shproj}]
```

Comment:
Why is this necessary?

Reply:
Just copy-pasting what is done on roslyn ;)